syntax="proto3";

package environment;

service environment {
    rpc createEnvironment(EnvironmentRequest) returns (EnvCreationResponse) {}
    rpc deleteEnvironment(EnvModReq) returns (Response) {}
    rpc stopEnvironment(EnvModReq) returns (Response) {}
}

message EnvironmentRequest {
    string name = 1;
    string tag = 2;
    string config = 3;
    repeated string images = 4;
    repeated string files = 5;
    string type = 6;
    string project_name = 7;
}

message EnvModReq {
    string id = 1;
}

message EnvCreationResponse {
    string message = 1;
    repeated string containers = 2;
    bool success = 3;
}

message Response {
    string message = 1;
}
